Source/WebCore/Modules/webaudio/AudioBufferSourceNode.cpp

```diff
@@ -241,9 +241,13 @@ bool AudioBufferSourceNode::renderFromBuffer(AudioBus* bus, unsigned destination
         virtualDeltaFrames = virtualMaxFrame - virtualMinFrame;
     }
 
-    // If we're looping and the offset (virtualReadIndex) is past the end of the loop, wrap back to the
-    // beginning of the loop. For other cases, nothing needs to be done.
-    if (m_isLooping && m_virtualReadIndex >= virtualMaxFrame) {
+    if (m_virtualReadIndex >= virtualMaxFrame) {
+        // Early exit to avoid going past the end of the source buffer.
+        if (!m_isLooping)
+            return false;
+
+        // If we're looping and the offset (virtualReadIndex) is past the end of the loop, wrap back to the
+        // beginning of the loop. For other cases, nothing needs to be done.
```

Comment:
Can we remove "For other cases, nothing needs to be done." here? With the comment inside the `if` statement, it no longer makes much sense nor does it seem like it needs to be said. I think the only part of the comment that is non-obvious, so worth writing, is "wrap back to the beginning of the loop". The rest is just a comment that says exactly what the code says ("if we’re looping" is the same as checking `m_isLooping`, past the end of the loop is the same thing as `m_virtualReadIndex >= virtualMaxFrame`), so not my favorite, although our web audio code has a lot of those kinds of comments. And note that before this code was documenting what was under it, now it’s referring to what’s above it, so less clear.

Reply:
Of course, I'll amend before landing, thanks! 🙏🏼